TESTING.asciidoc

```diff
@@ -142,6 +142,47 @@ Now the http.p12 file will be placed in the config directory of the running clus
 Assuming you have the http.ssl.keystore setup correctly, you can now use HTTPS with ./gradlew run without the risk
 of accidentally committing your local configurations.
 
+==== Multiple nodes in the test cluster for ./gradlew run
+
+Another desired customization for ./gradlew run might be to run multiple
+nodes with different setting for each node. For example, you may want to debug a coordinating only node that fans out
+to one or more data nodes. To do this, increase the numberOfNodes attribute for the test cluster configuration.
+-------------------------------------
+testClusters.register("runTask") {
+      ...
+      numberOfNodes = 2
+    }
+}
+-------------------------------------
+
+Then customize the run task itself to create different configurations per node. For example, the following will
+instruct the first node (:9200) to be a coordinating only node, and all other nodes to be master, data_hot, data_content
+nodes. Since we are changing the default master node configuration we must also update the default initial_master_nodes.
+Finally, we need to "restart" the cluster to pick up the changes. However, it does not actually start/stop/start
+the cluster since we have not started the cluster yet and this instructs the plugin to update the configuration.
+
+-------------------------------------
+tasks.register("run", RunTask) {
+  useCluster testClusters.named("runTask")
+  description = 'Runs elasticsearch in the foreground'
+  group = 'Verification'
+
+  doFirst {
+    def cluster = testClusters.named("runTask").get()
+    cluster.getNodes().each { node ->
+      node.setting('cluster.initial_master_nodes', cluster.getLastNode().getName())
+      node.setting('node.roles', '[master,data_hot,data_content]')
+    }
+    cluster.getFirstNode().setting('node.roles', '[]')
+    cluster.restart()
+  }
+  impliesSubProjects = true
```

Comment:
This setting is not required. This is just something to that running `./gradlew run` works from the CLI w/o specifyign the whole task path.

Reply:
I wondered about that setting , will remove it from the example (and replace the other non-relevant parts with `...` like in the other example.

Reply:
removed from example: [`e4649e5` (#89563)](https://github.com/elastic/elasticsearch/pull/89563/commits/e4649e50746bc28ceaecaba0f5cc71938f3e7157)